runtime/apps/apis/mailbox/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+// Licensed under the Apache-2.0 license
+
+//! # AsyncMailbox: Mailbox Interface
+
+use libtock_platform::allow_ro::{AllowRo, Config as ReadOnlyConfig};
+use libtock_platform::allow_rw::{AllowRw, Config as ReadWriteConfig};
+use libtock_platform::{share, DefaultConfig, ErrorCode, Syscalls};
+use libtockasync::TockSubscribe;
+/// Mailbox interface user interface.
+///
+/// # Generics
+/// - `DRIVER_NUM`: The driver number for the mailbox interface.
+/// - `S`: The syscall implementation.
+/// - `W`: Configuration for writable buffers (response).
+/// - `R`: Configuration for read-only buffers (input), defaults to `DefaultConfig`.
+pub struct AsyncMailbox<
+    const DRIVER_NUM: u32,
+    S: Syscalls,
+    W: ReadWriteConfig,
```

Comment:
If possible, I think we also want to eliminate these configs from the generics. I don't see us changing the values, and it makes things harder to read and write.